vec3d.h

```diff
@@ -0,0 +1,130 @@
+#include<math.h>
+
+
+template<typename T>
+class Vec3{
+    private:
+        T x, y, z;
+    public:
+        Vec3() : x(T(0)), y(T(0)), z(T(0)) {}           
+        Vec3(T xx) : x(xx), y(xx), z(xx) {}
+        Vec3(T xx, T yy, T zz): x(xx), y(yy), z(zz) {}
```

Comment:
As a minor side note: `Vec3(T x, T y, T z): x(x), y(y), z(z) {}` would also work. There is fear of variable shadowing here, and using the same name for the parameter and the member variables is standard practice.